### PR TITLE
Enable Anaconda build deps for other branches

### PR DESCRIPTION
**Pull request by jkonecny12**
In the past only Fedora 37 image had Anaconda dependencies. Let's change that and add the dependencies to all the branches.

This solution will try if Anaconda has a branch for given Fedora version in format "fedora-XX" and if that fails let's use Rawhide branch of Anaconda.
This solution is required mainly because Fedora Rawhide is name fedora-XX in the script.

 * [x] image-refresh fedora-38
 * [x] image-refresh fedora-rawhide

### CODE REVIEW

File: images/scripts/lib/build-deps.sh [hunk 1]
```diff
@@ -25,12 +25,16 @@ esac
...
-    $GET https://raw.githubusercontent.com/rhinstaller/anaconda/master/anaconda.spec.in | \
-        sed 's/@PACKAGE.*@/0/' | rpmspec --buildrequires --query /dev/stdin | \
-        sed 's/.*/"&"/' | tr '\n' ' '
-fi
+# We build anaconda rpms
+case "$OS_VER" in
+    fedora*) # let's try if Anaconda have branch for the given fedora, if not, use master (rawhide)
+        anaconda_branch="fedora-${OS_VER#fedora }"
+        $GET "https://raw.githubusercontent.com/rhinstaller/anaconda/$anaconda_branch/anaconda.spec.in" > /dev/null || anaconda_branch=master
+        $GET "https://raw.githubusercontent.com/rhinstaller/anaconda/$anaconda_branch/anaconda.spec.in" | \
```
Comment (martinpitt) [inline images/scripts/lib/build-deps.sh:33]: Seems a bit wasteful to run this twice -- I'd recommend something like
```sh
SPEC="$($GET ...anaconda_branch URL)" || SPEC="$($GET ...master branch URL)"
```

**Comment by martinpitt** [comment]
Triggered a fedora-38 image for now. Please let me know if you need other ones, or just edit the tasks yourself (but please not while an image build is ongoing, as  that will cancel it)

**Comment by cockpituous** [comment]
image-refresh fedora-38 done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-38-20230324-154419

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-38-20230324-154444.log

**Comment by jkonecny12** [comment]
Please update rawhide image too.

File: images/scripts/lib/build-deps.sh [hunk 2]
```diff
@@ -25,12 +25,17 @@ esac
...
-# We build anaconda rpms on fedora-37
-if [ "$OS_VER" = 'fedora 37' ]; then
-    $GET https://raw.githubusercontent.com/rhinstaller/anaconda/master/anaconda.spec.in | \
-        sed 's/@PACKAGE.*@/0/' | rpmspec --buildrequires --query /dev/stdin | \
-        sed 's/.*/"&"/' | tr '\n' ' '
-fi
+# We build anaconda rpms
+case "$OS_VER" in
+    fedora*) # let's try if Anaconda have branch for the given fedora, if not, use master (rawhide)
+        anaconda_branch="fedora-${OS_VER#fedora }"
```
Comment (martinpitt) [inline images/scripts/lib/build-deps.sh:31]: Unused variable. Would actually look a little cleaner if you used it below, but leaving that up to you.

File: images/scripts/lib/build-deps.sh [hunk 3]
```diff
@@ -25,12 +25,17 @@ esac
...
+        SPEC=$($GET "https://raw.githubusercontent.com/rhinstaller/anaconda/fedora-${OS_VER#fedora }/anaconda.spec.in") 2>/dev/null || \
```
Comment (martinpitt) [inline images/scripts/lib/build-deps.sh:32]: I'd drop the `2>/dev/null` -- it's easier for debugging to show the error message in the log.

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-rawhide-20230324-160808.log

File: images/scripts/lib/build-deps.sh (hunk 2 above)
Comment (jkonecny12) [reply images/scripts/lib/build-deps.sh]: Fixed, I'll remove it. Seems to me the same honestly.

File: images/scripts/lib/build-deps.sh (hunk 3 above)
Comment (jkonecny12) [reply images/scripts/lib/build-deps.sh]: Done.

**Comment by cockpituous** [comment]
image-refresh fedora-38 done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-38-20230325-063738

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-38-20230325-063737.log

**Comment by martinpitt** [comment]
I rebuilt f-38 to test the new changes, looks good. I pushed the images here and triggered tests. @jkonecny12, please let me know if they work for you!

**Review by martinpitt — APPROVED** [review]

**Comment by jkonecny12** [comment]
Seems like everything went well: https://github.com/rhinstaller/anaconda/pull/4622

Thanks!

**Pull request merged**